File: src/main/java/com/teamwizardry/librarianlib/api/gui/components/ComponentRaw.java
package com.teamwizardry.librarianlib.api.gui.components;

import java.util.function.Consumer;

import com.teamwizardry.librarianlib.api.gui.GuiComponent;
import com.teamwizardry.librarianlib.api.util.math.Vec2;

public class ComponentRaw extends GuiComponent<ComponentRaw> {
	
	public Consumer<ComponentRaw> func;
	
	public ComponentRaw(int posX, int posY, Consumer<ComponentRaw> func) {
		super(posX, posY);
		this.func = func;
	}

	public ComponentRaw(int posX, int posY, int width, int height, Consumer<ComponentRaw> func) {
		super(posX, posY, width, height);
		this.func = func;
	}

	@Override
	public void drawComponent(Vec2 mousePos, float partialTicks) {
		func.accept(this);
	}

}

File: src/main/java/com/teamwizardry/librarianlib/client/gui/components/ComponentRaw.kt
package com.teamwizardry.librarianlib.client.gui.components

import com.teamwizardry.librarianlib.client.gui.GuiComponent
import com.teamwizardry.librarianlib.common.util.math.Vec2d

import java.util.function.Consumer

class ComponentRaw : GuiComponent<ComponentRaw> {

    var func: Consumer<ComponentRaw>

    constructor(posX: Int, posY: Int, func: Consumer<ComponentRaw>) : super(posX, posY) {
        this.func = func
    }

    constructor(posX: Int, posY: Int, width: Int, height: Int, func: Consumer<ComponentRaw>) : super(posX, posY, width, height) {
        this.func = func
    }

    override fun drawComponent(mousePos: Vec2d, partialTicks: Float) {
        func.accept(this)
    }

}
